### PR TITLE
[Debug] Clean code in Qwen 3 Omni and add warning for talker temperature.

### PR DESCRIPTION
**Pull request by tzhouam**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose
As descibed in #687 and #494 , the code for Qwen 3 Omni needs to be cleaned, and the talker temperature should have warning if be set to 0. 
- Improve Qwen3Omni sampling behavior: consolidate zero-temperature warnings in the talker stage via a new _warn_talker_sampling_temperature, ensuring a single first-use warning and passing sampling_metadata through GPU runner compute_logits for correct sampling handling.

- Strengthen initialization and robustness: initialize tts_tokens in the thinker stage, add clearer error logging when TTS token embedding fails, and handle profile-only runs safely.

- Simplify/clean code: remove redundant logits processors and the submodule device-move helper to keep the model path leaner.

## Test Plan
Tested by offline
## Test Result
For talker temperature warning, if we have temperature less than 0, we have a warning:
```bash
(Worker pid=11999) [Stage-1] WARNING 01-07 09:27:32 [qwen3_omni.py:986] Temperature is set to None, as all requests are greedy. This is equivalent to setting temperature to 0.0.Please consider setting a higher temperature i.e. 0.4.
(Worker pid=11999) [Stage-1] WARNING 01-07 09:27:32 [qwen3_omni.py:986] This warning will be shown only once, for the first request where temperature is 0.0. Later requests will not show this warning but still be affected by the temperature.
(Worker pid=11999) [Stage-1] WARNING 01-07 09:27:32 [qwen3_omni_moe_code_predictor_mtp.py:196] Using sdpa attention backend (flash_attention_2 not available or failed)
(Worker pid=11914) [Stage-2] INFO 01-07 09:41:39 [mrope.py:452] Multimodal token idx changed!
INFO 01-07 09:41:41 [log_utils.py:549] {'type': 'request_level_metrics',
INFO 01-07 09:41:41 [log_utils.py:549]  'request_id': '0_14e55db1-6216-4c61-8f6f-8971874c5f90',
```
---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by hsliuustc0106** [comment]
fix ci please
